pkg/broker/receiver.go

```diff
@@ -41,9 +41,10 @@ const (
 
 // Receiver parses Cloud Events, determines if they pass a filter, and sends them to a subscriber.
 type Receiver struct {
-	logger   *zap.Logger
-	client   client.Client
-	ceClient cloudevents.Client
+	logger             *zap.Logger
+	client             client.Client
+	ceClient           cloudevents.Client
+	structuredCeClient cloudevents.Client
```

Comment:
it would be better to have a flag in the context that the transport could watch for an use the structured encoder for the same client. Just like target.